felles/sikkerhet/src/main/java/no/nav/vedtak/tokenx/TokenXchange.java

```diff
@@ -0,0 +1,49 @@
+package no.nav.vedtak.tokenx;
+
+
+import static no.nav.vedtak.log.util.ConfidentialMarkerFilter.CONFIDENTIAL;
+
+import java.net.URI;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import no.nav.foreldrepenger.konfig.Environment;
+import no.nav.vedtak.sikkerhet.oidc.config.OpenIDProvider;
+import no.nav.vedtak.sikkerhet.oidc.token.OpenIDToken;
+import no.nav.vedtak.sikkerhet.oidc.token.SikkerhetContext;
+import no.nav.vedtak.sikkerhet.oidc.token.TokenProvider;
+
+/**
+ * Dette filteret skal brukes når man vet man mottar et token som støtter
+ * on-behalf-of, og når mottakende system krever kun dette
+ *
+ */
+public final class TokenXchange {
+    private static final Environment ENV = Environment.current();
+    private static final Logger LOG = LoggerFactory.getLogger(TokenXchange.class);
+
+    private TokenXchange() {
+        // NOSONAR
+    }
+
+    public static OpenIDToken exchange(URI targetEndpoint) {
+        var token = TokenProvider.getTokenUtenSamlFallback(SikkerhetContext.BRUKER);
+        return exchange(token, targetEndpoint);
+    }
+
+    public static OpenIDToken exchange(OpenIDToken token, URI targetEndpoint) {
+        if (token != null && OpenIDProvider.TOKENX.equals(token.provider())) {
+            LOG.trace(CONFIDENTIAL, "Veksler tokenX token {} for {}", token, targetEndpoint);
+            var assertion = TokenXAssertionGenerator.assertion();
+            return TokenProvider.exchangeTokenX(token, assertion, targetEndpoint);
+        } else {
+            if (token != null && ENV.isLocal()) {
```

Comment:
Vi bør bruke https://github.com/navikt/mock-oauth2-server istedenfor snekre alt selv i vtp.

Reply:
Før hadde de ikke tilrettelagt for generering av token som er koblet til en bruker. Nå har de laget noe støtte før det i https://github.com/navikt/fakedings

Vi får se på det litt nærmere sammen med sikkerhetsarbeidet i Høst.